### PR TITLE
Add DNS provider for CIVO

### PR DESCRIPTION
**Pull request by dirien**
Hi,

This PR add [CIVO](https://civo.com/) support to the lego support!

Looking forward for you feedback.

### CODE REVIEW

**Comment by ldez** [comment]
Hello, in order for a PR adding a DNS provider to be accepted, you have to:

- [ ] add a description to your PR
- [ ] be able to maintain this provider
- [ ] have a homogeneous design with the other providers
- [ ] add tests (units)
```shell
make test
```
- [ ] add tests ("live")
https://github.com/go-acme/lego/blob/c938de68f262c057542d44bcb67ef961cefcc60e/providers/dns/cloudflare/cloudflare_test.go#L125-L151

```shell
make test
```
- [ ] add a [provider descriptor](https://github.com/go-acme/lego/blob/master/providers/dns/cloudflare/cloudflare.toml)
- [ ] generate CLI help, documentation, and readme.
```bash
make generate-dns
```
- [ ] be able to do: _(and put the output of this command to a comment in your PR)_
```bash
rm -rf .lego

./lego -m your@email.com --dns YOUR_PROVIDER_NAME -d *.example.com -d example.com -s https://acme-staging-v02.api.letsencrypt.org/directory run
```
Note the wildcard domain is important.
- [ ] pass the linter ([golangci-lint](https://github.com/golangci/golangci-lint#install) must be installed):
```shell
make checks
```
- [ ] do `go mod tidy`

File: providers/dns/civo/civo.toml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+[Configuration]
+    [Configuration.Credentials]
+        CIVO_TOKEN = "Authentication token"
+    [Configuration.Additional]
+        CIVO_POLLING_INTERVAL = "Time between DNS propagation check"
+        CIVO_PROPAGATION_TIMEOUT = "Maximum waiting time for DNS propagation"
+        CIVO_TTL = "The TTL of the TXT record used for the DNS challenge"
+
+[Links]
+    API = "https://www.civo.com/learn/configure-dns"
```
Comment (ldez) [inline providers/dns/civo/civo.toml:21]: Can we have a public link to the API documentation?
The current link is not API documentation.

**Review by ldez — CHANGES_REQUESTED** [review]

File: providers/dns/civo/civo.toml (hunk 1 above)
Comment (dirien) [reply providers/dns/civo/civo.toml:21]: Thanks for the hint! Added the API docs page

**Comment by dirien** [comment]
Output:

```bash
root@lively-harbor-85ea-f77859:/tmp# rm -rf .lego/
root@lively-harbor-85ea-f77859:/tmp# ./lego --email info@ediri.de --dns civo -d *.ediri.de -d ediri.de -s https://acme-staging-v02.api.letsencrypt.org/directory run
2022/09/01 13:06:42 No key found for account info@ediri.de. Generating a P256 key.
2022/09/01 13:06:42 Saved key to /tmp/.lego/accounts/acme-staging-v02.api.letsencrypt.org/info@ediri.de/keys/info@ediri.de.key
2022/09/01 13:06:43 Please review the TOS at https://letsencrypt.org/documents/LE-SA-v1.2-November-15-2017-w-v1.3-notice.pdf
Do you accept the TOS? Y/n
y
2022/09/01 13:06:47 [INFO] acme: Registering account for info@ediri.de
!!!! HEADS UP !!!!

Your account credentials have been saved in your Let's Encrypt
configuration directory at "/tmp/.lego/accounts".

You should make a secure backup of this folder now. This
configuration directory will also contain certificates and
private keys obtained from Let's Encrypt so making regular
backups of this folder is ideal.
2022/09/01 13:06:47 [INFO] [*.ediri.de, ediri.de] acme: Obtaining bundled SAN certificate
2022/09/01 13:06:48 [INFO] [*.ediri.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/3503921513
2022/09/01 13:06:48 [INFO] [ediri.de] AuthURL: https://acme-staging-v02.api.letsencrypt.org/acme/authz-v3/3503921523
2022/09/01 13:06:48 [INFO] [*.ediri.de] acme: use dns-01 solver
2022/09/01 13:06:48 [INFO] [ediri.de] acme: Could not find solver for: tls-alpn-01
2022/09/01 13:06:48 [INFO] [ediri.de] acme: Could not find solver for: http-01
2022/09/01 13:06:48 [INFO] [ediri.de] acme: use dns-01 solver
2022/09/01 13:06:48 [INFO] [*.ediri.de] acme: Preparing to solve DNS-01
2022/09/01 13:06:48 [INFO] [ediri.de] acme: Preparing to solve DNS-01
2022/09/01 13:06:48 [INFO] [*.ediri.de] acme: Trying to solve DNS-01
2022/09/01 13:06:48 [INFO] [*.ediri.de] acme: Checking DNS record propagation using [127.0.0.53:53]
2022/09/01 13:07:18 [INFO] Wait for propagation [timeout: 5m0s, interval: 30s]
2022/09/01 13:07:23 [INFO] [*.ediri.de] The server validated our request
2022/09/01 13:07:23 [INFO] [ediri.de] acme: Trying to solve DNS-01
2022/09/01 13:07:23 [INFO] [ediri.de] acme: Checking DNS record propagation using [127.0.0.53:53]
2022/09/01 13:07:53 [INFO] Wait for propagation [timeout: 5m0s, interval: 30s]
2022/09/01 13:08:00 [INFO] [ediri.de] The server validated our request
2022/09/01 13:08:00 [INFO] [*.ediri.de] acme: Cleaning DNS-01 challenge
2022/09/01 13:08:00 [INFO] [ediri.de] acme: Cleaning DNS-01 challenge
2022/09/01 13:08:00 [INFO] [*.ediri.de, ediri.de] acme: Validations succeeded; requesting certificates
2022/09/01 13:08:01 [INFO] [*.ediri.de] Server responded with a certificate.
```

**Comment by dirien** [comment]
Don't know how to solve 

> be able to maintain this provider

**Comment by ldez** [comment]
> Don't know how to solve

Who are you? what is your relation with CIVO?

**Comment by dirien** [comment]
> > Don't know how to solve
> 
> Who are you? what is your relation with CIVO?

A CIVO user.

**Comment by dirien** [comment]
End user but I will take care of the this and will let CIVO know about this!

**Review by ldez — CHANGES_REQUESTED** [review]
I have a problem with the CIVO client, I don't want to depend on the huge k8s stack.

```
k8s.io/api v0.21.2 // indirect
k8s.io/apimachinery v0.21.2 // indirect
k8s.io/client-go v0.21.2 // indirect
k8s.io/klog/v2 v2.8.0 // indirect
k8s.io/utils v0.0.0-20201110183641-67b214c5f920 // indirect
sigs.k8s.io/structured-merge-diff/v4 v4.1.0 // indirect
sigs.k8s.io/yaml v1.2.0 // indirect
```

So I will prefer a light client implementation inside lego.
ex: https://github.com/go-acme/lego/tree/master/providers/dns/constellix/internal

**Comment by dirien** [comment]
Hi @ldez,

okay np! Let me change this to use the API directly!

**Comment by dirien** [comment]
Created a PR in the `civogo` repo! It's a utility function they recently added and hopefully they can move it out again.

Here the link to my PR -> https://github.com/civo/civogo/pull/118

**Comment by dirien** [comment]
HI @ldez,

The good people at Civo merged my PR to remove the k8s libs. 

Looking forward for your feedback!

**Review by ldez — APPROVED** [review]
Thank you :+1:

**Pull request merged**